server/src-lib/Hasura/RQL/Types/ScheduledTrigger.hs

```diff
@@ -0,0 +1,103 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Hasura.RQL.Types.ScheduledTrigger
+  ( ScheduleType(..)
+  , CreateScheduledTrigger(..)
+  , RetryConfST(..)
+  , formatTime'
+  ) where
+
+import           Data.Time.Clock
+import           Data.Time.Clock.Units
+import           Data.Time.Format
+import           Data.Aeson
+import           Data.Aeson.Casing
+import           Data.Aeson.TH
+import           Hasura.Prelude
+import           System.Cron.Types
+import           System.Cron.Parser
+import           Hasura.Incremental
+
+import qualified Data.Text                     as T
+import qualified Data.Aeson                    as J
+import qualified Hasura.RQL.Types.EventTrigger as ET
+
+data RetryConfST
+  = RetryConfST
+  { rcstNumRetries  :: !Int
+  , rcstIntervalSec :: !Seconds
+  , rcstTimeoutSec  :: !Seconds
+  , rcstTolerance   :: !NominalDiffTime
```

Comment:
I think it probably makes sense to use united types for all three of these fields or none of them, but the mixture here is confusing. I’ve noticed that this type has a derived `FromJSON` instance, so presumably the serialization is important… but I don’t actually know what the `ToJSON`/`FromJSON` instances for `NominalDiffTime` actually does. Is it exposed in an API? And if so, does it make sense from a user’s POV?

Reply:
Considering `DiffTime` and `NominalDiffTime` can be negative (see here: https://github.com/hasura/graphql-engine/pull/3553/files#r379279442), should we change these to `PostiveInt` or something?